src/components/Modal/index.tsx

```diff
@@ -20,12 +20,14 @@ const Modal: React.FC<Props> = props => {
       }
     };
 
-    document.addEventListener('mousedown', handleClickOutside);
+    open
+      ? document.addEventListener('mousedown', handleClickOutside)
+      : document.removeEventListener('mousedown', handleClickOutside);
 
     return () => {
       document.removeEventListener('mousedown', handleClickOutside);
```

Comment:
@viniciusaa same here, remove the event listener through checking the `open` property.

Reply:
@paulohfev,  I can't remove this. This is necessary to remove the event listener when the component is unmounted.